src/main/java/duke/Task.java

```diff
@@ -0,0 +1,45 @@
+package duke;
+
+/**
+ * Represent a Task.
+ */
+public class Task {
+    protected final String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Retrieves the status of the task.
+     * @return X if the task is done or a whitespace if it is not done.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); //mark done task with X
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Unmark the task as done.
+     */
+    public void unMarkAsDone() {
```

Comment:
I could understand that the naming unMarkAsDone somewhat gives it a better contrast with markAsDone haha, but perhaps need to change to unmarkAsDone() since unmark is a word itself.